pybamm/models/submodels/porosity/full_reaction_driven_porosity.py

```diff
@@ -17,8 +17,9 @@ class Full(BaseModel):
     **Extends:** :class:`pybamm.porosity.BaseModel`
     """
 
-    def __init__(self, param):
-        super().__init__(param)
+    def __init__(self, param, options):
+        super().__init__(param, options)
+        self.options = options
```

Comment:
If you make the change I suggested in `base_porosity.py` I think this is no longer needed (`base_submodel.py` will take care of it).